packages/core/src/actions/waitForTaskCompletionWs.ts

```diff
@@ -0,0 +1,48 @@
+import axios from 'axios'
+import type { Config } from '../createConfig.js'
+import { getTaskStatus } from './getTaskStatus.js'
+import type { WaitForTaskCompletionParameters } from './waitForTaskCompletion.js'
+import type { RelayerWebsocketMessage } from '../types/ws.js'
+import { TASK_STATUS_ROUTE } from '../constants.js'
+import { websocketWaiter } from '../utils/websocketWaiter.js'
+
+export async function waitForTaskCompletionWs(
+  config: Config,
+  parameters: WaitForTaskCompletionParameters,
+): Promise<null | undefined> {
+  const { id, timeout } = parameters
+
+  const url = config.getWebsocketBaseUrl()
+  const topic = TASK_STATUS_ROUTE(id)
+
+  const prefetch = async () => {
+    try {
+      await getTaskStatus(config, { id })
+    } catch (error) {
+      if (axios.isAxiosError(error)) {
+        if (error.response?.status === 404) {
+          // Assume a 404 means the task is completed
+          return null
+        }
+      }
```

Comment:
Is this assumption still valid? I added this check before we had task history in the relayer, and have wanted to remove this if we could just check for historical task status.

Reply:
i'll double check, but i'm pretty sure this still holds, and what's nice about this approach is it works for _all_ tasks - some are not tracked in history so checking task history doesn't work for everything

Reply:
yeah, this assumption still holds - it's pretty safe to use assuming you are using a task ID returned by another API call, i.e. that of a task you know to have been created on the relayer. if you pass in a random UUID you'll get a false positive, but i can just call that out in a doc comment

Reply:
per internal discussion, going to make the changes relayer-side to ensure that all tasks that may reasonably be queried by an API client are tracked in task history, so we will use the task history endpoint for this prefetch